cmd/cosign/cli/verify/verify_blob.go

```diff
@@ -161,252 +149,109 @@ func (c *VerifyBlobCmd) Exec(ctx context.Context, blobRef string) error {
-	case c.BundlePath != "":
+	}
+	if c.BundlePath != "" {
 		b, err := cosign.FetchLocalSignedPayloadFromPath(c.BundlePath)
 		if err != nil {
 			return err
 		}
-		if b.Cert == "" {
+		// A certificate is required in the bundle unless we specified with
+		//  --key, --sk, or --certificate.
+		if b.Cert == "" && co.SigVerifier == nil && cert == nil {
 			return fmt.Errorf("bundle does not contain cert for verification, please provide public key")
 		}
-		// b.Cert can either be a certificate or public key
-		certBytes := []byte(b.Cert)
-		if isb64(certBytes) {
-			certBytes, _ = base64.StdEncoding.DecodeString(b.Cert)
-		}
-		cert, err = loadCertFromPEM(certBytes)
-		if err != nil {
-			// check if cert is actually a public key
-			co.SigVerifier, err = sigs.LoadPublicKeyRaw(certBytes, crypto.SHA256)
-		} else {
-			if c.CertChain == "" {
-				co.RootCerts, err = fulcio.GetRoots()
-				if err != nil {
-					return fmt.Errorf("getting Fulcio roots: %w", err)
-				}
-				co.IntermediateCerts, err = fulcio.GetIntermediates()
-				if err != nil {
-					return fmt.Errorf("getting Fulcio intermediates: %w", err)
-				}
-				co.SigVerifier, err = cosign.ValidateAndUnpackCert(cert, co)
-				if err != nil {
-					return fmt.Errorf("verifying certificate from bundle: %w", err)
-				}
-			} else {
-				// Verify certificate with chain
-				chain, err := loadCertChainFromFileOrURL(c.CertChain)
-				if err != nil {
-					return err
-				}
-				co.SigVerifier, err = cosign.ValidateAndUnpackCertWithChain(cert, chain, co)
+		// We have to condition on this because sign-blob may not output the signing
+		// key to the bundle when there is no tlog upload.
+		if b.Cert != "" {
+			// b.Cert can either be a certificate or public key
+			certBytes := []byte(b.Cert)
+			if isb64(certBytes) {
+				certBytes, _ = base64.StdEncoding.DecodeString(b.Cert)
+			}
+			cert, err = loadCertFromPEM(certBytes)
+			if err != nil {
+				// check if cert is actually a public key
+				co.SigVerifier, err = sigs.LoadPublicKeyRaw(certBytes, crypto.SHA256)
 				if err != nil {
-					return fmt.Errorf("verifying certificate from bundle with chain: %w", err)
+					return fmt.Errorf("loading verifier from bundle: %w", err)
 				}
 			}
 		}
+		opts = append(opts, static.WithBundle(b.Bundle))
+	}
+	// Set an SCT if provided via the CLI.
+	if c.SCTRef != "" {
+		sct, err := os.ReadFile(filepath.Clean(c.SCTRef))
 		if err != nil {
-			return fmt.Errorf("loading verifier from bundle: %w", err)
+			return fmt.Errorf("reading sct from file: %w", err)
 		}
-		bundle = b.Bundle
-	default:
-		return fmt.Errorf("please provide a cert to verify against via --certificate or a bundle via --bundle")
+		co.SCT = sct
 	}
-
-	// Performs all blob verification.
-	if err := verifyBlob(ctx, co, blobBytes, sig, cert, bundle); err != nil {
-		return err
+	// Set a cert chain if provided.
+	var chainPEM []byte
+	if c.CertChain != "" {
+		chain, err := loadCertChainFromFileOrURL(c.CertChain)
+		if err != nil {
```

Comment:
nit: if chain == nil, that seems like an error to me, because we expect a chain

Reply:
we can do that